server/controllers/likeDislike.js

```diff
@@ -0,0 +1,167 @@
+    }
+
+    if ((reaction === 'like') !== true) {
+      if ((reaction === 'dislike') !== true) {
+        return res.status(400).json({
+          status: 'error',
+          message: "Reaction must either be 'like' or 'dislike'",
+        });
+      }
+    }
+
+    const status = reaction === 'like' ? '1' : '0';
+
+    Article.findOne({
+      where: {
+        slug: articleSlug,
+      }
+    })
+      .then((article) => {
+        if (!article) {
+          return res.status(404).json({
+            status: 'error',
+            message: 'Article was not found.',
+          });
+        }
+        const articleId = article.id;
+
+        LikesDislikes.findOne({
+          where: {
+            userId,
+          }
+        })
+          .then((likeOrDislike) => {
+            if (!likeOrDislike) {
+              LikesDislikes.create({
+                articleId,
+                userId,
+                reaction: status,
+              })
+                .then(() => res.status(200).json({
+                  status: 'success',
+                  message: `Article ${reaction}d successfully.`,
+                }))
+                .catch(err => next(err));
+            } else if (likeOrDislike.reaction === status) {
+              LikesDislikes.destroy({
+                where: {
+                  articleId,
+                  userId,
+                }
+              })
+                .then(() => res.status(200).json({
+                  status: 'success',
+                  message: 'Reaction removed.',
+                })).catch(err => next(err));
+            } else {
+              LikesDislikes.update({
+                reaction: status,
+              }, {
+                where: {
+                  userId,
+                  articleId,
+                }
+              })
+                .then(() => res.status(200).json({
+                  status: 'success',
+                  message: `Article ${reaction}d successfully.`,
+                }))
+                .catch(err => next(err));
+            }
+          }).catch(err => next(err));
+      }).catch(err => next(err));
+  }
+
+  /**
+   * @desc Get all reaction for an article
+   * @param {obj} req request body
+   * @param {obj} res response body
+   * @param {obj} next next action
+   * @return {obj} returns an object
+   */
+  static getLikesDislikes(req, res, next) {
+    const { articleSlug } = req.params;
+    const reactions = { likes: 0, dislikes: 0 };
+    if (!articleSlug || articleSlug.trim().length < 1) {
+      return res.status(400).json({
+        status: 'error',
+        message: 'Please provide a valid article slug.',
+      });
+    }
+
+    return Article.findOne({
+      where: {
+        slug: articleSlug,
+      }
+    })
+      .then((article) => {
+        if (!article) {
+          return res.status(404).json({
+            status: 'error',
+            message: 'Article was not found.',
+          });
+        }
+        LikesDislikes.findAll({
```

Comment:
let likes,
let dislikes
LikesDislikes.findAll({
where: {
            articleId: article.id,
            reaction: 0}).then((allLikes) => {if (!allLikes} { likes === 0} else {likes = allLikes.Length})


same for dislikes. return data: {likes}, {dislikes}
          },